test/recipes/15-test_dhparam.t

```diff
@@ -0,0 +1,97 @@
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:3072 -pkeyopt qbits:256 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:0 -out $TESTDIR/p3072_q256_t1862_pgen0.pem
+
+
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:1024 -pkeyopt qbits:160 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p1024_q160_t1862_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:1024 -pkeyopt qbits:224 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p1024_q224_t1862_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:1024 -pkeyopt qbits:256 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p1024_q256_t1862_pgen1.pem
+
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:1024 -pkeyopt qbits:160 -pkeyopt type:fips186_4 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p1024_q160_t1864_pgen1.pem
+
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:2048 -pkeyopt qbits:160 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p2048_q160_t1862_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:2048 -pkeyopt qbits:224 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p2048_q224_t1862_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:2048 -pkeyopt qbits:256 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p2048_q256_t1862_pgen1.pem
+
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:2048 -pkeyopt qbits:224 -pkeyopt type:fips186_4 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p2048_q224_t1864_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:2048 -pkeyopt qbits:256 -pkeyopt type:fips186_4 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p2048_q256_t1864_pgen1.pem
+
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:3072 -pkeyopt qbits:160 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p3072_q160_t1862_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:3072 -pkeyopt qbits:224 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p3072_q224_t1862_pgen1.pem
+./util/opensslwrap.sh genpkey -genparam -algorithm DH -pkeyopt pbits:3072 -pkeyopt qbits:256 -pkeyopt type:fips186_2 -pkeyopt dh_paramgen_type:1 -out $TESTDIR/p3072_q256_t1862_pgen1.pem
+
+=cut
+
+my @valid = glob(data_file("valid", "*.pem"));
+#my @invalid = glob(data_file("invalid", "*.pem"));
+
+my $num_tests = scalar @valid;# + scalar @invalid;
+plan tests => 2 * $num_tests;
+
+ SKIP: {
+    skip "Skipping DH tests", $num_tests
+        if disabled('deprecated-3.0');
+
+    foreach (@valid) {
+        ok(run(app([qw{openssl dhparam -noout -check -in}, $_])));
+    }
+
+#    foreach (@invalid) {
+#        ok(!run(app([qw{openssl dhparam -noout -check -in}, $_])));
+#    }
```

Comment:
Why this commented out code?

Reply:
It's a remnant of the original code copied from `test_ecparam`. If we get a set of invalid files, we could just uncomment it.